### PR TITLE
Dockerfile + updated lifecycle how to test

### PR DESCRIPTION
**Pull request by vire**
* added pushstate-server to serve compiled app

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+RUN cd /tmp &&\
+    yarn &&\
+    mkdir /app &&\
+    mv node_modules /app/node_modules &&\
+    rm package.json
+
+WORKDIR /app
+COPY . /app
+
+ARG REACT_APP_BACKEND_URL # env variable to specify API URL durin build
```
Comment (rlaucik) [inline Dockerfile]: had to remove comment from this line to make it work (otherwise I've got an error: "ARG requires exactly one argument definition")

Comment (vire) [reply Dockerfile]: Thanks for the hint! I've added the comment to tell why I'm passing a variable `REACT_APP_BACKEND_URL` during build time...

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
+FROM mhart/alpine-node:6
+MAINTAINER code.vire@gmail.com
+
+RUN npm install -g yarn
+
+COPY package.json /tmp
+RUN cd /tmp &&\
+    yarn &&\
+    mkdir /app &&\
+    mv node_modules /app/node_modules &&\
```
Comment (devopsmariocom) [inline Dockerfile:12]: why not run yarn directly inside of `/app` folder?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    rm package.json
+
+WORKDIR /app
+COPY . /app
```
Comment (devopsmariocom) [inline Dockerfile:16]: why not copy app in step when you are copying package.json?

**Review by devopsmariocom — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (vire) [reply Dockerfile:12]: This RUN will be cached so if I don't change `package.json` docker will use this layer from cache regardless of src code changes `/app`

Comment (vire) [reply Dockerfile:12]: I think you're right could be simplified maybe in one step, I can try it out

Comment (devopsmariocom) [reply Dockerfile:12]: yep not sure why they came up with intermediate tmp folder, maybe it didn't work before ¯_(ツ)_/¯

File: Dockerfile (hunk 3 above)
Comment (devopsmariocom) [reply Dockerfile:16]: I'm not sure about your exact use case but your are right with catching but as it's only local catching I don't see it as much valuable but YMMV.

Also any reason to not use yarn.lock file to ensure exact packages versions?

Comment (vire) [reply Dockerfile:16]: I wasn't sure about putting yarn.lock right there until 1.0 of yarn, but ths comment conviced me!

**Review by rlaucik — APPROVED** [review]

**Pull request merged**